src/vision/roadmap/polish/sync_and_async.md

```diff
@@ -15,3 +15,56 @@ unexpected should happen.
 | ---                                      | ---    | ---              |
 | Define "behave the same"                 | 💤     |  |
 | Create testing to ensure same behavior   | 💤     |  |
+
+## Details
+
+Ideally, there should not be a lot of work needed specifically to achieve this goal.
+Instead, the primary aim is to define principles that can inform the design in other places.
+That said, automated testing to verify we have achieved these principles may take significant effort.
+
+There are several ways we can look at what it means to behave the same.
+One way is from a language and semantics standpoint, while another is from a library and ecosystem standpoint.
+We will look at each of these in more detail, and then lay out some ideas for automated testing.
+
+### Language and Semantics
+
+Roughly want we want here is that code that differs only in its syncness should do the same thing.
+Of course, this is not strictly possible because a sync and async program are fundamentally different program.
```

Comment:
Also: would it be worth slightly elaborating _how_ these models differ? I strongly suspect that async Rust is strictly a superset of non-async Rust, specifically because it enables concurrent operation and cancellation, with all other capabilities flowing from these properties.

Reply:
Good catch. I meant "are fundamentally different programs," not so much about the model although that would be interesting to be more precise about. What I was trying to convey here is that if you make a change to a program, it is now a different program so it will have different behavior. So the goal is to come up with some kind of equivalence that says what things can change and what things should not.